apps/examples/src/examples/zoom-to-bounds/ZoomToBoundsExample.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState } from 'react'
+import { Box, createShapeId, Tldraw, toRichText, useEditor, Vec } from 'tldraw'
+import './zoom-to-bounds.css'
+
+// There is a guide at the bottom of this file :)
+
+function CameraButtons({
+	box1,
+	box2,
+	initialCameraLockedState,
+}: {
+	box1: Box
+	box2: Box
+	initialCameraLockedState: boolean
+}) {
+	const editor = useEditor()
+	const [isCameraLocked, setIsCameraLocked] = useState(initialCameraLockedState)
+
+	const toggleCameraLock = () => {
+		const newLockedState = !isCameraLocked
+		setIsCameraLocked(newLockedState)
+		editor.setCameraOptions({ isLocked: newLockedState })
+	}
+
+	// [1]
+	const customZoomToBounds = (
+		box: Box,
+		forceZoomEvenIfLocked: boolean = false,
+		animationDuration: number
+	) => {
+		editor.zoomToBounds(box, {
+			force: forceZoomEvenIfLocked,
+			inset: 32,
+			animation: { duration: animationDuration },
+		})
+	}
+
+	// [2]
+	return (
+		<div className="camera-controls">
+			{/* [a] */}
+			<button
+				onClick={toggleCameraLock}
+				className={`camera-button camera-button--lock ${isCameraLocked ? 'locked' : ''}`}
+			>
+				{isCameraLocked ? '🔒 Unlock Camera' : '🔓 Lock Camera'}
+			</button>
+
+			{/* [b] */}
+			<button
+				onClick={() => customZoomToBounds(box1, false, 750)}
+				disabled={isCameraLocked}
+				className="camera-button camera-button--zoom"
+			>
+				Zoom to Box 1
+			</button>
+
+			{/* [c] */}
+			<button
+				onClick={() => customZoomToBounds(box2, true, 100)}
+				className="camera-button camera-button--zoom"
+			>
+				Zoom to Box 2 (Forced)
+			</button>
+		</div>
+	)
+}
+
+export default function ZoomToBoundsExample() {
+	const position1 = new Vec(1000, 500)
+	const size1 = new Vec(500, 400)
+
+	const position2 = new Vec(50, 100)
+	const size2 = new Vec(900, 720)
```

Comment:
Because these positions are getting hard-coded here, if you move either of the boxes, it moves the camera to a blank space.